### ISSUE TITLE
collisions due to overlapping loads

### ISSUE THREAD

**Issue by svpv** [Cyan4973/xxHash#259]
There are massive collisions on ELF symbols (which are functions names found in binaries and shared libraries).
```
05c3bcfaf53bd8d1816832569ec6482f        imdi_k488_gentab
05c3bcfaf53bd8d1816832569ec6482f        imdi_k48_gentab
075c7212a88df56ff9390d0bb4098e23        elput_init
075c7212a88df56ff9390d0bb4098e23        elput_input_init
```
I pushed the [full list](https://raw.githubusercontent.com/svpv/xxh3-collisions/master/collisions.list) along with the [data set](https://github.com/svpv/xxh3-collisions) (`elfsym0u.zst`, about 9 million strings).

**Comment by Cyan4973** [comment]
Thanks for these tests and feedback @svpv .
It seems to happen in the 9-16 input range, which gives a clear target to analyze.

**Comment by Cyan4973** [comment]
Come on ....
There is a clear mistake in this module : it doesn't count the contribution of input length anywhere in the formula. This is clearly an oversight. I'll fix it, I expect it to solve the reported collision issue all by itself.

Another issue is: how could this mistake manage to pass the tests ? I must have another look at the test suite, it must catch these cases.

**Comment by easyaspi314** [comment]
Ouch. 

Yup, that's a blooper.

```c
static XXH128_hash_t XXH3_len_9to16_128b(BYTE const *XXH_RESTRICT input,
                                      size_t len,
                                      BYTE const *XXH_RESTRICT secret,
                                      XXH64_hash_t seed)
{
    U128 acc = 
        (U128)(XXH_readLE64(secret) + seed)
      | ((U128)(XXH_readLE64(secret + 8) - seed) << 64);
    acc ^=
        (XXH_readLE64(input)
       |((U128)XXH_readLE64(input + len - 8) << 64);
    acc ^= (acc >> 64);
    acc *= PRIME64_1;
    acc ^= (acc >> 96);
    acc *= PRIME64_2;
    {
        XXH128_hash_t ret;
        ret.low64 = XXH3_avalanche((U64)acc);
        ret.high64 = XXH3_avalanche((U64)(acc >> 64));
        return ret;
    }
}
```

**Comment by Cyan4973** [comment]
Fix ongoing in #260.

I still need to work on these test tools.
But running these tests can be long, and I might need a quick release next week for internal usages.

**Comment by Cyan4973** [comment]
So, `SMHasher` is unable to produce this sequence on input,
and my own collision analyzer only works one same-size samples.

So I guess I'm going to create a new tool to catch those ones...

**Comment by Cyan4973** [comment]
It's more difficult than I anticipated 
to modify a generic pattern generator
to output similar-yet-unique patterns like the ones reported in this issue.
It's a pity, as a generic method could have been merged into the collision analyzer tool.

Another method would be to create a dedicated pattern generator,
which would be written just to generate patterns similar to those reported by @svpv.
Might be simpler and faster ...
